client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ListenerCacheManagerStopTest.java

```diff
@@ -0,0 +1,118 @@
+ * Test to verify that the listener threads are stopped.
+ *
+ * @author gustavonalle
+ * @since 7.2
+ */
+@Test(groups = "functional", testName = "client.hotrod.event.ListenerStopTest")
+public class ListenerCacheManagerStopTest extends AbstractInfinispanTest {
+
+   HotRodServer hotRodServer;
+   RemoteCacheManager remoteCacheManager;
+   RemoteCache<Integer, String> cache;
+
+   @BeforeMethod
+   public void setup() throws IOException {
+      Integer port = getRandomFreePort();
+      hotRodServer = HotRodTestingUtil.startHotRodServer(createCacheManager(hotRodCacheConfiguration()), port);
+      remoteCacheManager = new RemoteCacheManager(
+            new ConfigurationBuilder()
+                  .addServer()
+                  .port(port)
+                  .host(HotRodTestingUtil.host())
+                  .connectionTimeout(3000)
+                  .socketTimeout(3000)
+                  .build()
+      );
+      cache = remoteCacheManager.getCache();
+   }
+
+   @AfterMethod
+   public void tearDown() {
+      HotRodClientTestingUtil.killServers(hotRodServer);
+      HotRodClientTestingUtil.killRemoteCacheManager(remoteCacheManager);
+   }
+
+   @Test
+   public void testThreadsAreStopped() throws Exception {
+      final EventLogListener listener = new EventLogListener();
+      cache.addClientListener(listener);
+
+      final String listenerId = findListenerId(listener);
+      assertListenerThreadRunning(listenerId);
+
+      remoteCacheManager.stop();
+      hotRodServer.stop();
+
+      assertListenerThreadNotRunning(listenerId);
+   }
+
+   private String findListenerId(Object listener) {
+      ClientListenerNotifier clientNotifier = extractField(remoteCacheManager, "listenerNotifier");
+      return Util.toHexString(clientNotifier.findListenerId(listener), 8);
+
+   }
+
+   private void assertListenerThreadRunning(final String listenerId) {
+      eventually(new Condition() {
+         @Override
+         public boolean isSatisfied() throws Exception {
+            return isListenerThreadRunning(listenerId);
+         }
+      });
+   }
+
+   private void assertListenerThreadNotRunning(final String listenerId) {
+      eventually(new Condition() {
+         @Override
+         public boolean isSatisfied() throws Exception {
+            return !isListenerThreadRunning(listenerId);
+         }
+      });
+   }
+
+   private boolean isListenerThreadRunning(String listenerId) {
+      ThreadMXBean threadMXBean = ManagementFactory.getThreadMXBean();
+      ThreadInfo[] threadInfos = threadMXBean.dumpAllThreads(false, false);
+      for (ThreadInfo threadInfo : threadInfos) {
+         if ((threadInfo.getThreadState().equals(Thread.State.RUNNABLE) ||
+                    threadInfo.getThreadState().equals(Thread.State.BLOCKED)) &&
+               threadInfo.getThreadName().contains(listenerId)) {
+            return true;
+         }
+      }
+      return false;
+   }
+
+   private Integer getRandomFreePort() throws IOException {
```

Comment:
Nice trick but not sure it was necessary, since `HotRodTestingUtil.startHotRodServer` is overloaded to avoid taking a port, generating it for you.